docs/design/BucketReplication.md

```diff
@@ -0,0 +1,140 @@
+# Bucket Replication
+
+
+## GOALS
+ * data replication of objects which are stored in any MCG bucket (data bucket, namespace bucket) -
+    * Provides a higher resiliency
+    * Data buckets / namespace buckets can be backed by any supported cloud provider (S3, Azure etc)
+    * The replication is one way replication - from a source MCG bucket to a destination MCG bucket.
+    * Replication can be bidirectional by setting a complementing replication configuration on the destination bucket.
+ * provide the user the ability to set a replication configuration on a bucket. 
+
+## OUT OF SCOPE
+ * Versions are not replicated - only the latest version
+ * Deletions are not replicated
+ * Optimizations
+
+
+## SOLUTION
+
+<div id="top" />
+<img src="/docs/design/images/bucket_replication.png" />
+
+### Bucket replication policy schema
+  * a replication policy composed of a list of replication rules,
+    each rule defines rule id and a destination bucket 
+    and optionally defines a filter based on the object key prefix
+
+Structure of replication policy which was added to bucket document in bucket collection:
+
+```
+    replication_policy: {
+        type: 'array',
+        items: {
+            type: 'object',
+            required: ['destination_bucket', 'rule_id'],
```

Comment:
I'm not sure I understand the purpose of the `rule_id`

Reply:
it's for compatibility with S3 bucket replication